examples/core/rgcn/hetero_rgcn.py

```diff
@@ -0,0 +1,647 @@
+        sampler,
+        batch_size=16384,
+        shuffle=False,
+        num_workers=0,
+        device=device,
+    )
+
+    # Initialize a progress bar for visual feedback.
+    pbar = tqdm(total=y_true.size(0))
+    pbar.set_description(f"Inference")
+
+    # To store the predictions.
+    y_hats = list()
+
+    for input_nodes, seeds, blocks in loader:
+        blocks = [blk.to(device) for blk in blocks]
+        # We only predict the nodes with type "category".
+        seeds = seeds[category]
+        batch_size = seeds.shape[0]
+        input_nodes_indexes = input_nodes[category].to(g.device)
+
+        # Extract node embeddings for the input nodes.
+        emb = extract_embed(node_embed, input_nodes)
+        # Add the batch's raw "paper" features.
+        # Corresponds to the content in the function `rel_graph_embed` comment.
+        emb.update({category: g.ndata["feat"][category][input_nodes_indexes]})
+        emb = {k: e.to(device) for k, e in emb.items()}
+
+        # Generate predictions.
+        logits = model(emb, blocks)[category]
+        # Apply softmax to the logits and get the prediction by selecting the
+        # argmax.
+        y_hat = logits.log_softmax(dim=-1).argmax(dim=1, keepdims=True)
+        y_hats.append(y_hat.cpu())
+
+        pbar.update(batch_size)
+
+    pbar.close()
+
+    y_pred = th.cat(y_hats, dim=0)
+    y_true = th.unsqueeze(y_true, 1)
+
+    # Calculate the accuracy of the predictions for the train, valid and
+    # test splits.
+    train_acc = evaluator.eval(
+        {
+            "y_true": y_true[split_idx["train"]["paper"]],
+            "y_pred": y_pred[split_idx["train"]["paper"]],
+        }
+    )["acc"]
+    valid_acc = evaluator.eval(
+        {
+            "y_true": y_true[split_idx["valid"]["paper"]],
+            "y_pred": y_pred[split_idx["valid"]["paper"]],
+        }
+    )["acc"]
+    test_acc = evaluator.eval(
+        {
+            "y_true": y_true[split_idx["test"]["paper"]],
+            "y_pred": y_pred[split_idx["test"]["paper"]],
+        }
+    )["acc"]
+
+    return train_acc, valid_acc, test_acc
+
+
+def is_support_affinity(v_t):
+    # dgl supports enable_cpu_affinity since 0.9.1.
+    return v_t >= "0.9.1"
+
+
+def main(args):
+    device = f"cuda:0" if th.cuda.is_available() else "cpu"
+
+    # Prepare the data.
+    g, labels, num_classes, split_idx, logger, train_loader = prepare_data(
+        args, device
+    )
+
+    # Create the embedding layer and move it to the appropriate device.
+    embed_layer = rel_graph_embed(g, 128).to(device)
+
+    # Initialize the entity classification model.
+    model = EntityClassify(g, 128, num_classes).to(device)
+
+    print(
+        "Number of embedding parameters: "
+        f"{sum(p.numel() for p in embed_layer.parameters())}"
+    )
+    print(
+        "Number of model parameters: "
+        f"{sum(p.numel() for p in model.parameters())}"
+    )
+
+    for run in range(args.runs):
```

Comment:
why we need multiple runs and none of the homograph node classification supports multiple runs, is it special to heterograph? Or just for this example or just for ogbn?

Reply:
I've noticed that the examples under OGB are all tested multiple times and calculate the average value. In some other parts, some are run multiple times while others are not. There seems to be no uniform style in this regard.